aggregator/operator_registrations_inmemory.go

```diff
@@ -0,0 +1,190 @@
+				if err != nil {
+					ors.logger.Error("Error opening websocket subscription for new pubkey registrations", "err", err, "service", "OperatorRegistrationsServiceInMemory")
+					// see the warning above the struct definition to understand why we panic here
+					panic(err)
+				}
+
+			case newPubkeyRegistrationEvent := <-newPubkeyRegistrationC:
+				pubkeys := types.OperatorPubkeys{
+					G1Pubkey: bls.NewG1Point(newPubkeyRegistrationEvent.PubkeyG1.X, newPubkeyRegistrationEvent.PubkeyG1.Y),
+					G2Pubkey: bls.NewG2Point(newPubkeyRegistrationEvent.PubkeyG2.X, newPubkeyRegistrationEvent.PubkeyG2.Y),
+				}
+				operatorId := types.OperatorIdFromPubkey(pubkeys.G1Pubkey)
+				operatorAddr := newPubkeyRegistrationEvent.Operator
+
+				pubkeyByAddrDict[operatorAddr] = pubkeys
+				pubkeyByIdDict[operatorId] = pubkeys
+
+				ors.logger.Debug("Added operator pubkeys to pubkey dict",
+					"service", "OperatorRegistrationsServiceInMemory",
+					"block", newPubkeyRegistrationEvent.Raw.BlockNumber,
+					"operatorAddr", operatorAddr,
+					"operatorId", operatorId,
+					"G1pubkey", pubkeyByAddrDict[operatorAddr].G1Pubkey,
+					"G2pubkey", pubkeyByAddrDict[operatorAddr].G2Pubkey,
+				)
+
+			// Receive a queryByAddr from GetOperatorPubkeys
+			case operatorPubkeyQuery := <-queryByAddrC:
+				pubkeys, ok := pubkeyByAddrDict[operatorPubkeyQuery.operatorAddr]
+				operatorPubkeyQuery.respC <- resp{pubkeys, ok}
+
+			// Receive a queryById from GetOperatorPubkeysById
+			case operatorPubkeyQuery := <-queryByIdC:
+				pubkeys, ok := pubkeyByIdDict[operatorPubkeyQuery.operatorId]
+				operatorPubkeyQuery.respC <- resp{pubkeys, ok}
+			}
+		}
+	}()
+}
+
+func (ors *OperatorRegistrationsServiceInMemory) queryPastRegisteredOperatorEventsAndFillDb(ctx context.Context, pubkeyByAddrDict map[common.Address]types.OperatorPubkeys, pubkeyByIdDict map[types.OperatorId]types.OperatorPubkeys) {
+	// Querying with nil startBlock and stopBlock will return all events. It doesn't matter if we queryByAddr some events that we will receive again in the websocket,
+	// since we will just overwrite the pubkey dict with the same values.
+	alreadyRegisteredOperatorAddrs, alreadyRegisteredOperatorPubkeys, err := ors.avsRegistryReader.QueryExistingRegisteredOperatorPubKeys(ctx, nil, nil)
+	if err != nil {
+		ors.logger.Error("Fatal error querying existing registered operators", "err", err, "service", "OperatorRegistrationsServiceInMemory")
+		panic(err)
+	}
+
+	ors.logger.Debug("List of queried operator registration events in blsApkRegistry", "alreadyRegisteredOperatorAddr", alreadyRegisteredOperatorAddrs, "service", "OperatorRegistrationsServiceInMemory")
+	for i, operatorAddr := range alreadyRegisteredOperatorAddrs {
+		operatorPubkeys := alreadyRegisteredOperatorPubkeys[i]
+		pubkeyByAddrDict[operatorAddr] = operatorPubkeys
+
+		operatorId := types.OperatorIdFromPubkey(operatorPubkeys.G1Pubkey)
+		pubkeyByIdDict[operatorId] = operatorPubkeys
+	}
```

Comment:
I would just inline this code, it provides no value to the reader to have it separated from the only call site

Reply:
You mean inline this function in `startServiceInGoroutine`? It provides value by simplifying reader's flow in main logic block within startServiceInGoroutine by slicing it into execution of smaller logic blocks, so one doesn't have to read the whole thing in one function.